include/mxnet/gc.h

```diff
@@ -0,0 +1,138 @@
+/*
```

Comment:
This doesn't need to be under include.
put it in src/kvstore and rename to gradient_compression

Reply:
When including this from kvstore.h, the include would be weird. 
#include "../../src/kvstore/gradient_compression.h" 

Is that ok?